src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -556,13 +556,14 @@ private BoundExpression MakeImplicitConversion(BoundExpression rewrittenOperand,
 
                 TypeSymbol userDefinedConversionRewrittenType = conversion.Method.ReturnType;
 
-                if (rewrittenOperand.Type != conversion.Method.ParameterTypes[0])
+                // the conversion only needs to happen for non-nullable valuetypes
+                if (rewrittenOperand.Type != conversion.Method.ParameterTypes[0] && !userDefinedConversionRewrittenType.CanBeAssignedNull())
                 {
                     Debug.Assert(rewrittenOperand.Type.IsNullableType());
                     Debug.Assert(rewrittenOperand.Type.GetNullableUnderlyingType() == conversion.Method.ParameterTypes[0]);
-                    Debug.Assert(!userDefinedConversionRewrittenType.IsNullableType());
 
                     // Lifted conversion, wrap return type in Nullable
+                    // BUG: Construct(...) should check the conversion is correct, as it is now, it will not check generic type parameter constraints
```

Comment:
Rather than adding this comment, I think we should use more specific and precise condition for the ```if```. If I interpret the intent of this code correctly its purpose is to set ```userDefinedConversionRewrittenType``` to result type of a lifted conversion operator, if we are dealing with a lifted operator. So, let's use condition that checks for that explicitly.
```
rewrittenOperand.Type.IsNullableType() && 
conversion.Method.ParameterTypes[0] == rewrittenOperand.Type.GetNullableUnderlyingType() &&
!userDefinedConversionRewrittenType.IsNullableType() && 
userDefinedConversionRewrittenType.IsValueType
```